eodatasets3/assemble.py

```diff
@@ -846,12 +857,52 @@ def write_thumbnail(
             percentile_stretch=percentile_stretch,
             input_geobox=grid,
         )
-        self._checksum.add_file(thumb_path)
 
-        accessory_name = "thumbnail"
-        if kind:
-            accessory_name += f":{kind}"
-        self.add_accessory_file(accessory_name, thumb_path)
+        self._document_thumbnail(thumb_path, kind)
+
+    def write_thumbnail_singleband(
+        self,
+        measurement: str,
+        thumb_path,
+        bit: int = None,
+        lookup_table: Dict[str, Tuple[int, int, int]] = None,
+        kind: str = None,
+    ):
+        """
+        Write a singleband thumbnail out, taking in an input measurement and
+        outputting a JPG with appropriate settings.
+
+        Options are to
+        EITHER
+        Use a bit (int) as the value to scale from black to white to
+        i.e., 0 will be BLACK and bit will be WHITE, with a linear scale between.
+        OR
+        Provide a lookuptable (dict) of int (key) [R, G, B] (value) fields
+        to make the image with.
+        """
+
+        thumb_path = self.names.thumbnail_name(self._work_path, kind=kind)
+
+        _, image_path = self.measurements.get(measurement, None)
```

Comment:
The unpacking will always fail when measurement is None, so we'll never get to the below validation.

```suggestion
        _, image_path = self.measurements.get(measurement, (None, None))
```